### DISCUSSION TITLE
Does latest openssl plugins compatible with older version of strongswan code  ? If yes, will the IKE micro service is compatible with it ?

### DISCUSSION THREAD

**Discussion by shrinivasad** [strongswan/strongswan#1550] [General]
I'm planning to use latest openssl 3.0.6 plugins in older version of strongswan.

Using these crypto and ssl libraries to run IKE micro service. This is giving errors.

I'm using boost library or timer expiry operation in the micro service.

Error:

terminate called after throwing an instance of 'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::log::v2s_mt_posix::system_error> >'dco6_ike_remote_ms    |   what():  Failed to set TLS value: Invalid argumentdco6_ike_remote_ms    | terminate called recursively

**Comment by tobiasbrunner** [comment]
What do you mean with "IKE micro service"?

strongSwan's _openssl_ plugin is only really compatible with OpenSSL 3.x since [5.9.6](https://github.com/strongswan/strongswan/releases/tag/5.9.6). And while it might be possible to use a later version of the plugin with an older version of strongSwan, I wouldn't recommend it. Why would you want to use an old version of strongSwan anyway?

Note that strongSwan only needs libcrypto, libssl is not linked.

**Reply by shrinivasad** [reply]
Thanks for the reply. My requirement is to use older version of strongswan.

And the issue I'm facing is with the boost library for latest openssl.

Does boost ASIO support openssl 3.0 ?

**Reply by tobiasbrunner** [reply]
> My requirement is to use older version of strongswan.

Why?

> And the issue I'm facing is with the boost library for latest openssl.
> 
> Does boost ASIO support openssl 3.0 ?

I've absolutely no idea but it's clearly unrelated to strongSwan.
